src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -335,8 +293,8 @@ public void setErrorMessageHandler(ErrorMessageHandler errorMessageHandler) {
     }
 
     private void waitForState(ReadyState target) {
-        long iterations = options.getConnectionTimeoutMillis() / 100;
-        while (errorWhileConnecting == null && getState() != target && iterations > 0) {
+        long iterations = (options.getReconnectRetryInterval()  + options.getConnectionTimeoutMillis() + 500) / 100;
+        while (getState() != target && iterations > 0) {
```

Comment:
The wait time changes from `getConnectionTimeoutMillis()` to `getConnectionTimeoutMillis() + getReconnectRetryInterval()` here, making connection timeouts longer than expected from user perspective. (I don't think it matters though, so doesn't necessarily need to be addressed). 

I'm assuming this is because we don't know when the next scheduled check will be, so we assume the maximum. This could be somewhat improved if we knew _when_ it will be (can it be queried from the executor?).

Another note, readability of this section would IMO be improved if we conditioned on the time waited instead of on its derivative, `iterations`:

```
long timeWaited = 0;
while (getState() != target && timeWaited < maxWaitTime) {
    ...
    Thread.sleep(100);
    timeWaited += 100;
    ...
}

Reply:
Agree'd. I cleaned up a bit as you suggested.